dev_suites/dev_demo_ig_stu1/demo_suite.rb

```diff
@@ -116,6 +116,31 @@ class DemoSuite < Inferno::TestSuite
       end
     end
 
+    group do
+      id 'hidden_group'
+      title 'Hidden Inputs Group'
+      description 'This group contains 2 inputs that are optional and hidden from the UI.'
+
+      optional
+
+      test 'Hidden, optional, empty input test' do
+        input :hidden_optional_empty, title: 'Hidden and Optional (should be runnable)',
+                                      description: 'Example of hidden, empty, optional field',
+                                      hidden: true,
+                                      optional: true
+        run { pass }
+      end
+
+      test 'Hidden, optional, filled input test' do
+        input :hidden_optional_filled, title: 'Hidden and Optional (should be runnable)',
+                                       description: 'Example of hidden, filled, optional field',
+                                       default: 'example text',
+                                       hidden: true,
+                                       optional: true
+        run { pass }
```

Comment:
Rather than just passing, this should verify that it is receiving the expected value from the front end.

We should also have an example of hiding a field in an authinfo input so that we can demonstrate that working.

Reply:
updated. now getting error from frontend when adding optional & hidden input to auth info field
<img width="1296" alt="Screenshot 2025-03-13 at 1 03 24 PM" src="https://github.com/user-attachments/assets/e3fcf477-633e-4fc9-80ce-63033a93cff6" />
<img width="1296" alt="Screenshot 2025-03-13 at 1 04 19 PM" src="https://github.com/user-attachments/assets/562a30af-7f5a-4a0d-8c39-719ee95c2f4b" />

Reply:
I think that's ok for now, when the backend part is looking good, we can have Alyssa make the front end changes.

I was thinking some more about this last night, and I think another case we should allow is hiding locked inputs, even if they're required. Particularly for AuthInfo, I could imagine wanting to hide particular locked fields in order to reduce clutter.

Reply:
makes sense. updated